### ISSUE TITLE
AsyncLocalStorage not available on web, so stackblitz fails

### ISSUE THREAD

**Issue by nksaraf** [solidjs/solid-start#1172]
I haven't been able to run the [basic example of start 0.4 in stackblitz](https://stackblitz.com/github/solidjs/solid-start/tree/main/examples/basic) as is (no prerender flags), because I get this error:

```sh
[h3] [unhandled] TypeError: Cannot read properties of undefined (reading 'assets')
    at get assets [as assets] (/home/projects/pariwmzoo.github/node_modules/.pnpm/@solidjs+start@0.4.2_solid-js@1.8.7_vinxi@0.0.54_vite@5.0.10/node_modules/@solidjs/start/server/StartServer.tsx:50:109)
```

_Originally posted by @birkskyum in https://github.com/solidjs/solid-start/issues/1162#issuecomment-1867086436_

**Comment by albertoforni** [comment]
I have a similar problem trying to migrate from `v0.3.9` to `v0.4.2`

`__vite-browser-external:node:async_hooks:3  Uncaught (in promise) Error: Module "node:async_hooks" has been externalized for browser compatibility. Cannot access "node:async_hooks.AsyncLocalStorage" in client code.`

It's imported from here https://github.com/solidjs/solid/blob/main/packages/solid/web/storage/storage.ts#L1

I see this error when running `pnpm run dev`

<img width="810" alt="image" src="https://github.com/solidjs/solid-start/assets/2559136/514ebc59-934b-45a9-8cbd-d8bda350bac2">

**Comment by ryansolid** [comment]
I'm going to try to partially polyfill it. Isn't completely polyfillable but at least for like 80% usage it should work. THe idea here is in contexts that are async `getRequestEvent` should work until the first await in every environment. It will work anywhere in envs that support ALS though.

**Comment by birkskyum** [comment]
@ryansolid , that is very much appreciated, thanks! I'd love to keep debugging the static builds, but it's just too challenging without the ability to share repros, because we get stuck in "Works on my machine".

**Comment by ryansolid** [comment]
Ok released in 0.4.3.. there are gaps but the examples all work. Basically you have to be careful where you call getRequestEvent but it otherwise works.
